File: designpatterns/tic-tac-toeImproved/src/com/techlabs/mark/Game.java
package com.techlabs.mark;

public class Game {
	 private Player player1;
	 private Player player2;
	 private Board  board;
	 private BoardAnalyser anlyser;
	 
	
	public Game(Player player1, Player player2, Board board) {
	    
		this.player1=player1;
		this.player2=player2;
		this.board=board;
        this.anlyser= new BoardAnalyser(this.board);  		
	}
	
	public void move(Player player , int position) {
         if(position==1) {
        	 board.markPlayerSign(0, 0, player.getSign());
         }else if(position==2) {
        	 board.markPlayerSign(0, 1, player.getSign());
         }else if(position==3) {
        	 board.markPlayerSign(0, 2, player.getSign());
         }else if(position==4) {
        	 board.markPlayerSign(1, 0, player.getSign());
         }else if(position==5) {
        	 board.markPlayerSign(1, 2, player.getSign());
         }else if(position==6) {
        	 board.markPlayerSign(1, 3, player.getSign());
         }else if(position==7) {
        	 board.markPlayerSign(2, 0, player.getSign());
         }else if(position==8) {
        	 board.markPlayerSign(2, 1, player.getSign());
         }else if(position==9) {
        	 board.markPlayerSign(2, 2, player.getSign());
         }
         
         
		
	}
 
}

File: nodejs/tic-tac-toe/game.js
const readlinesync = require('readline-sync')
require('console-sync')

module.exports = class game {

    constructor(board, player1, player2) {
        this.board = board
        this.player1 = player1
        this.player2 = player2
    }
    get Board() {
        return this.board
    }
    get Player1() {
        return this.player1
    }
    get Player2() {
        return this.player2
    }
    gameStart() {
        const readinput = readlinesync;
        var term = 8
        var playerflag = true;
        while (term != 0) {
            let rowindex
            let columindex
            let player;

            console.log("player" + (Number(term % 2) + Number(1)) + " move")
            rowindex = readinput.question('where do you want insert mark in row')

            columindex = readinput.question('where do you want insert mark in column')
            if (playerflag == true) {
                player = this.player1
            } else {
                player = this.player2
            }
            var success = this.board.getCelltoMark(rowindex, columindex, player.PlayerMark)
            if (success == true) {
                term--;
                if (playerflag == false) {
                    playerflag = true
                } else {
                    playerflag = false
                }
            } else console.log("game End")


            console.log(this.board.getBoardMarks())
        }
    }

}